lib/chai/core/assertions.js

```diff
@@ -466,10 +740,26 @@ module.exports = function (chai, _) {
   /**
    * ### .NaN
    *
-   * Asserts that the target is `NaN`.
+   * Asserts that the target is exactly `NaN`.
    *
    *     expect(NaN).to.be.NaN;
-   *     expect(4).not.to.be.NaN;
+   *
+   * In modern environments, this assertion is performed using the built-in ES6
+   * `Number.isNaN` method. In legacy environments, the target is compared to
+   * itself using strict (`===`) equality. The result is the same either way:
+   * the target is only considered to be `NaN` if it's exactly `NaN`.
```

Comment:
🛑 I feel like this is not relevant to the user here. If there is a genuine edge case that users should be wary of, we should consider it a bug and fix the edge case.